app/views/handlers/question.py

```diff
@@ -195,12 +195,15 @@ def handle_post(self):
         # pylint: disable=no-member
         # wtforms Form parents are not discoverable in the 2.3.3 implementation
         self.questionnaire_store_updater.update_answers(self.form.data)
-        self.questionnaire_store_updater.update_progress_for_dependent_sections()
         if self.questionnaire_store_updater.is_dirty():
+            # We prematurely complete the block, as we need it completed to build the routing path
+            # In order to support progress value source references of the previous block
+            self.questionnaire_store_updater.add_completed_location()
             self._routing_path = self.router.routing_path(
                 section_id=self._current_location.section_id,
                 list_item_id=self._current_location.list_item_id,
             )
+        self.questionnaire_store_updater.update_progress_for_dependent_sections()
```

Comment:
This is evaluating path twice since it is called here then also in `block.py` as `super().handle_post()` will call it.

Reply:
This one is a bit tricky because it because when I remove this it breaks the test for `test_happy_path_then_make_calculated_summary_incomplete`, I think because the update needs to happen before the section completeness is updated in handle post 😐

Reply:
I have removed this now, but it seems in order for all tests cases to pass, the update needs to happen both before and after section completeness is updated, which i'm not sure is correct 😕

Reply:
Ya need to work out why, as at the moment, it would be evaluating the progress for all the dependent sections twice. I'm not sure why we need to update the dependent sections before updating the status of the current section.